### PR TITLE
[structural binding][7/N] Replace std::tie with structural binding 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #120353

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/sparse/cuda/SoftMax.cu [hunk 1]
```diff
@@ -575,10 +566,10 @@ void cuda_sparse_coo_softmax_backward(
   if (nvalues > 0 && pool_size > 0) {
     cuda_sparse_coo_softmax_backward_kernel<scalar_t, LogSoftMax>
         <<<grid_size, block_size, 0, stream>>>(
-            sorted_indices.data_ptr<int64_t>(),
+            sorted_indices.template data_ptr<int64_t>(),
             pool_size,
-            pool_sizes.data_ptr<int64_t>(),
-            pool_offsets.data_ptr<int64_t>(),
+            pool_sizes.template data_ptr<int64_t>(),
+            pool_offsets.template data_ptr<int64_t>(),
```
Comment (Skylion007) [inline aten/src/ATen/native/sparse/cuda/SoftMax.cu:572]: I hope this doesn't increase code size much.

Comment (cyyever) [reply aten/src/ATen/native/sparse/cuda/SoftMax.cu:572]: Why?

File: aten/src/ATen/native/transformers/cuda/attention_backward.cu [hunk 2]
```diff
@@ -797,7 +795,7 @@ std::tuple<at::Tensor, at::Tensor, at::Tensor, at::Tensor> _scaled_dot_product_e
   sdp::CustomMaskType custom_mask_type = causal
     ? sdp::CustomMaskType::CausalFromTopLeft
     : sdp::CustomMaskType::NoCustomMask;
-  std::tie(grad_q, grad_k, grad_v, grad_bias) =
+  auto [grad_q, grad_k, grad_v, grad_bias] =
```
[the remark this answers is not in the document]
Comment (cyyever) [reply aten/src/ATen/native/transformers/cuda/attention_backward.cu:798]: Why?

[the remark this answers is not in the document]
Comment (Skylion007) [reply aten/src/ATen/native/transformers/cuda/attention_backward.cu:798]: Given the failing tests? Or are those just flakes?

[the remark this answers is not in the document]
Comment (cyyever) [reply aten/src/ATen/native/transformers/cuda/attention_backward.cu:798]: I

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**